ClLike/cl_like/cl_like.py

```diff
@@ -322,19 +324,38 @@ def _get_nz(self, cosmo, name, **pars):
         Applies shift and width nuisance parameters if needed.
         """
         z = self.bin_properties[name]['z_fid']
-        nz = self.bin_properties[name]['nz_fid']
+        nz = interp1d(z, self.bin_properties[name]['nz_fid'], kind='cubic',
+                      bounds_error=False, fill_value=(0, 'extrapolate'))
         zm = self.bin_properties[name]['zmean_fid']
         dz = 0.
         wz = 1.
+        jacob = 1
         if (self.nz_model == 'NzShift') or (self.nz_model == 'NzShiftWidth'):
             dz = pars.get(self.input_params_prefix + '_' + name + '_dz', 0.)
         if (self.nz_model == 'NzShiftWidth') or (self.nz_model == 'NzWidth'):
             wz = pars.get(self.input_params_prefix + '_' + name + '_wz', 1.)
-        z = zm+dz+(z-zm)/wz
-        msk = z >= 0
-        z = z[msk]
-        nz = nz[msk]
-        return (z, nz)
+            jacob = wz
+        # NzShiftParam parametrized as z_true - z_false = f(z_true)
+        if (self.nz_model == 'NzShiftParamExp'):
```

Comment:
do you think we need all of them? Just for simplicity, should we keep whatever you used for the actual paper? Up to you!

Reply:
I've removed the Exp case that we did not use in the end.